src/main/java/com/limechain/network/protocol/sync/SyncProtocol.java

```diff
@@ -26,36 +21,23 @@ public SyncProtocol() {
         super(MAX_REQUEST_SIZE, MAX_RESPONSE_SIZE);
     }
 
-    private static <T extends ProtocolMessageHandler<ByteBuf> & SyncController> CompletableFuture<SyncController>
-    onStartStream(Stream stream, T handler) {
+    @Override
+    protected CompletableFuture<SyncController> onStartInitiator(Stream stream) {
         stream.pushHandler(new ProtobufVarint32FrameDecoder());
+        stream.pushHandler(new ProtobufDecoder(SyncMessage.BlockResponse.getDefaultInstance()));
 
         stream.pushHandler((new ProtobufVarint32LengthFieldPrepender()));
         stream.pushHandler((new ProtobufEncoder()));
 
+        Sender handler = new Sender(stream);
         stream.pushHandler(handler);
         return CompletableFuture.completedFuture(handler);
     }
 
-    @Override
-    @NotNull
-    protected CompletableFuture<SyncController> onStartInitiator(@NotNull Stream stream) {
-        return onStartStream(stream, new Sender(stream));
-    }
-
-    @Override
-    @NotNull
-    protected CompletableFuture<SyncController> onStartResponder(@NotNull Stream stream) {
-        return onStartStream(stream, new Receiver(stream));
-    }
-
-    // Class for handling outgoing requests
-    @Log
-    static class Sender
-            implements ProtocolMessageHandler<ByteBuf>,
+    static class Sender implements ProtocolMessageHandler<SyncMessage.BlockResponse>,
```

Comment:
```suggestion
    static class SyncHandler implements ProtocolMessageHandler<SyncMessage.BlockResponse>,
```

We could change the name here or keep the name `Receiver` in the handler used by the StateProtocol. This would help keep a uniform manner of naming across protocols.

Reply:
I changed the StateProtocol's handler to Sender, as in future we will have req receiver protocols as well.